glue/viewers/image/composite_array.py

```diff
@@ -146,16 +147,12 @@ def __call__(self, bounds=None):
 
                 # ensure "bad" values have the same alpha as the
                 # rest of the layer:
-                if hasattr(layer['cmap'], 'get_bad'):
-                    bad_rgba = layer['cmap'].get_bad().tolist()
+                cmap_has_bad = hasattr(layer['cmap'], 'get_bad')
+                if cmap_has_bad:
+                    bad_rgba = layer['cmap'].get_bad()
                     bad_color = bad_rgba[:3]
-                    bad_alpha = bad_rgba[3:]
-
-                    if self._allow_bad_alpha:
-                        bad_rgba = bad_color + bad_alpha
-                    else:
-                        bad_rgba = bad_color + [layer['alpha']]
-
+                    bad_alpha = bad_rgba[-1] if self.cmap_bad_alpha is None else self.cmap_bad_alpha
+                    bad_rgba = bad_color + (bad_alpha,)
```

Comment:
Looks like ``bad_color`` is a [0,0,0] array at this point so adding 1 to it will give [1,1,1] rather than (0,0,0,1)

```suggestion
                    bad_rgba = tuple(bad_color) + (bad_alpha,)
```